### PR TITLE
Use Postgres to store tenants and users

### PR DESCRIPTION
**Pull request by metalmatze**
Instead of mocking the database with a config file and creating a bunch of in-memory indexes, we can simply use Postgres straightaway.

More details should be explained in the `DEVELOPMENT.md` directly. 

/cc @ant31 @brancz

### CODE REVIEW

File: database/migrations/1550596476_create_users_table.up.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+CREATE EXTENSION IF NOT EXISTS pgcrypto;
+
+CREATE TABLE users
+(
+  id       UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+  username VARCHAR(100) NOT NULL,
+  password TEXT         NOT NULL,
+  token    UUID             DEFAULT gen_random_uuid()
```
Comment (brancz) [inline database/migrations/1550596476_create_users_table.up.sql:8]: @ant31 wanted the token to be on the tenant level, as these are likely going to be used for machine authentication and shouldn't stop working when a user is deleted.

**Comment by brancz** [comment]
just one comment, otherwise lgtm

File: database/migrations/1550596476_create_users_table.up.sql (hunk 1 above)
Comment (metalmatze) [reply database/migrations/1550596476_create_users_table.up.sql:8]: Makes sense. Seems like that info didn't reach me.

Comment (metalmatze) [reply database/migrations/1550596476_create_users_table.up.sql:8]: Can we do this in a follow up PR. As this changes quite some things, including the authentication which then needs to happen on a tenant basis.

Comment (brancz) [reply database/migrations/1550596476_create_users_table.up.sql:8]: Sounds good to me

**Pull request merged**